src/NHibernate/Linq/NhLinqExpression.cs

```diff
@@ -87,6 +87,9 @@ public IASTNode Translate(ISessionFactoryImplementor sessionFactory, bool filter
 
 			var requiredHqlParameters = new List<NamedParameterDescriptor>();
 			var queryModel = NhRelinqQueryParser.Parse(_expression);
+			// Use the modified TransparentIdentifierRemovingExpressionVisitor in order to support expressions that were created
+			// programmatically and do not mimic what the C# compiler generates.
```

Comment:
```suggestion
			// programmatically and do not mimic what the C# compiler generates.
			// Consider removing this if https://re-motion.atlassian.net/projects/RMLNQ/issues/RMLNQ-121
			// is fixed and we upgrade ReLinq.
```

But maybe it should be added in `TransparentIdentifierRemovingExpressionVisitor` class comments instead.

Reply:
Moved to `TransparentIdentifierRemovingExpressionVisitor`.